api/src/controllers/courses.ts

```diff
@@ -33,18 +33,13 @@ router.post('/api/batch', (req: Request<{}, {}, { courses: string[] }>, res) =>
     res.json({});
   }
   else {
-    let r = fetch(process.env.PUBLIC_API_GRAPHQL_URL, {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json'
-      },
-      body: JSON.stringify({
-        query: getCourseQuery(req.body.courses)
-      })
-    });
-
-    r.then((response) => response.json())
-      .then((data) => res.json(data.data))
+    Promise.all(
+        req.body.courses.map(
+            (course) => fetch(process.env.PUBLIC_API_URL + 'courses/' + course)
+            .then((r) => r.json())
+            .then((r) => r.payload ? [r.payload.id, r.payload] : [])
+        )
+    ).then((data) => res.json(Object.fromEntries(data)));
```

Comment:
Yeah it does feel a bit slower, I will have to look into how best to resolve this.

The reason I used a batch of REST queries rather than a single composite GraphQL query (like before) is because currently the entire GraphQL query fails if a course does not exist. This can be resolved on the API end by making the `course` query return a nullable course; if that ends up being better performance wise, I'll update prod to reflect this behavior.

Reply:
Got it. To clarify, I assume this is occurring on the course page when a course that no longer exists is present in the prerequisite tree. (e.g. CSE45C as a prerequisite for ICS46). Or are there other scenarios where we request information for a nonexistent course (I assume our search is fine and websoc-fuzzy-search only returns existing courses)?

Reply:
Yeah pretty much; fuzzy search should only return courses that we have data for.